### PR TITLE
runs prometheus and uses gzip for the metrics route

### PR DESCRIPTION
**Pull request by amprokop**
Adds prometheus to our Rails servers, enabled only on development and demo.

Testing:
`/metrics` route spits out http request info
image

`certifications/new/123` isn't gzipped
image

`/metrics` is gzipped
image

### CODE REVIEW

**Comment by amprokop** [comment]
One note—how should we protect this route?

**Comment by askldjd** [comment]
Interesting question... I didn't think about this...

**Comment by askldjd** [comment]
Only two solutions comes to mind since we don't have a proxy that can protect this route.

1. Use pushgateway
2. Use prometheus [auth mechanism](https://prometheus.io/docs/operating/configuration/)
  - basic_auth
  - authorization header
3. Add nginx proxies to block out metrics from port 3000, and create a separate nginx proxy for `/metrics`. This way, we can let security group protect the endpoint.

File: config.ru [hunk 1]
```diff
@@ -1,4 +1,21 @@
 # This file is used by Rack-based servers to start the application.
 
 require ::File.expand_path("../config/environment", __FILE__)
+require "rack"
+require "prometheus/client/rack/collector"
+require "prometheus/client/rack/exporter"
+
+if Rails.env.development? || Rails.env.demo?
```
Comment (askldjd) [inline config.ru]: :+1: Thanks!

**Pull request merged**